python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+            for iIn in range(nIn):
+                coeffs[iOut, iIn] = baseCoeff * (iIn + 1) + coeffOffset
+                coeffs[iOut, iIn] *= inArray[iIn]
+        assert coeffs.ndim == 2
+        assert coeffs.shape == (nOut, nIn)
+        # Avoid spurious errors when comparing to a simplified array
+        return coeffs
+
+    def checkTransformation(self, transform, mapping, msg=""):
+        """Check tranForward and tranInverse for a transform
+
+        Parameters
+        ----------
+        transform : Transform
+            The transform to check
+        mapping : astshim.Mapping
+            The mapping the transform should use. This mapping
+            must contain valid forward or inverse transformations,
+            but they need not match if both present. Hence the
+            mappings returned by make*PolyMap are acceptable.
+        msg : string
+            Error message suffix describing test parameters
+        """
+        fromEndpoint = transform.getFromEndpoint()
+        toEndpoint = transform.getToEndpoint()
+        frameSet = transform.getFrameSet()
+
+        nIn = mapping.getNin()
+        nOut = mapping.getNout()
+        self.assertEqual(nIn, fromEndpoint.getNAxes(), msg=msg)
+        self.assertEqual(nOut, toEndpoint.getNAxes(), msg=msg)
+
+        # forward transformation of one point
+        rawInPoint = self.makeRawPointData(nIn)
+        inPoint = fromEndpoint.pointFromData(rawInPoint)
+
+        # forward transformation of an array of points
+        nPoints = 7  # arbitrary
+        rawInArray = self.makeRawArrayData(nPoints, nIn)
+        inArray = fromEndpoint.arrayFromData(rawInArray)
+
+        if mapping.hasForward():
+            self.assertTrue(transform.hasForward())
+            outPoint = transform.tranForward(inPoint)
+            rawOutPoint = toEndpoint.dataFromPoint(outPoint)
+            assert_allclose(rawOutPoint, mapping.tranForward(rawInPoint), err_msg=msg)
+            assert_allclose(rawOutPoint, frameSet.tranForward(rawInPoint), err_msg=msg)
+
+            outArray = transform.tranForward(inArray)
+            rawOutArray = toEndpoint.dataFromArray(outArray)
+            self.assertFloatsAlmostEqual(rawOutArray, mapping.tranForward(rawInArray), msg=msg)
+            self.assertFloatsAlmostEqual(rawOutArray, frameSet.tranForward(rawInArray), msg=msg)
+        else:
+            # Need outPoint, but don't need it to be consistent with inPoint
+            rawOutPoint = self.makeRawPointData(nOut)
+            outPoint = toEndpoint.pointFromData(rawOutPoint)
+            rawOutArray = self.makeRawArrayData(nPoints, nOut)
+            outArray = toEndpoint.arrayFromData(rawOutArray)
+
+            self.assertFalse(transform.hasForward())
+
+        if mapping.hasInverse():
+            self.assertTrue(transform.hasInverse())
+            # inverse transformation of one point;
+            # remember that the inverse will not give the original values!
```

Comment:
It is described in the description of the `mapping` parameter of this method. I added a pointer